pinot-plugins/pinot-batch-ingestion/v0_deprecated/pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/preprocess/OrcDataPreprocessingHelper.java

```diff
@@ -0,0 +1,233 @@
+        if (timestampColumnVector.noNulls || !timestampColumnVector.isNull[0]) {
+          return Long.toString(timestampColumnVector.time[0]);
+        } else {
+          return null;
+        }
+      case FLOAT:
+        // Extract to Float
+        DoubleColumnVector doubleColumnVector = (DoubleColumnVector) columnVector;
+        if (doubleColumnVector.noNulls || !doubleColumnVector.isNull[0]) {
+          return Float.toString((float) doubleColumnVector.vector[0]);
+        } else {
+          return null;
+        }
+      case DOUBLE:
+        // Extract to Double
+        doubleColumnVector = (DoubleColumnVector) columnVector;
+        if (doubleColumnVector.noNulls || !doubleColumnVector.isNull[0]) {
+          return Double.toString(doubleColumnVector.vector[0]);
+        } else {
+          return null;
+        }
+      case STRING:
+      case VARCHAR:
+      case CHAR:
+        // Extract to String
+        BytesColumnVector bytesColumnVector = (BytesColumnVector) columnVector;
+        if (bytesColumnVector.noNulls || !bytesColumnVector.isNull[0]) {
+          int length = bytesColumnVector.length[0];
+          return new String(bytesColumnVector.vector[0], bytesColumnVector.start[0], length, UTF_8);
+        } else {
+          return null;
+        }
+      case BINARY:
+        // Extract to byte[]
+        bytesColumnVector = (BytesColumnVector) columnVector;
+        if (bytesColumnVector.noNulls || !bytesColumnVector.isNull[0]) {
+          int length = bytesColumnVector.length[0];
+          byte[] bytes = new byte[length];
+          System.arraycopy(bytesColumnVector.vector[0], bytesColumnVector.start[0], bytes, 0, length);
+          return new String(bytes, StandardCharsets.UTF_8);
+        } else {
+          return null;
+        }
+      default:
+        // Unsupported types
+        throw new IllegalStateException("Unsupported field type: " + category + " for field: " + field);
+    }
+  }
+
+  /**
+   * Finds the orc file and return its orc schema.
+   */
+  private TypeDescription getOrcSchema(Path orcFile) {
+    TypeDescription orcSchema;
+    try (Reader reader = OrcFile.createReader(orcFile, OrcFile.readerOptions(HadoopUtils.DEFAULT_CONFIGURATION))) {
+      orcSchema = reader.getSchema();
+    } catch (Exception e) {
+      throw new IllegalStateException("Caught exception while extracting ORC schema from file: " + orcFile, e);
+    }
+    return orcSchema;
+  }
+
+  private void validateConfigsAgainstSchema(TypeDescription schema) {
+    List<String> fieldNames = schema.getFieldNames();
+    if (_partitionColumn != null) {
+      Preconditions.checkArgument(fieldNames.contains(_partitionColumn),
+          String.format("Partition column: %s is not found from the schema of input files.", _partitionColumn));
+      Preconditions.checkArgument(_numPartitions > 0,
+          String.format("Number of partitions should be positive. Current value: %s", _numPartitions));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: _numPartitions. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 217. Consider formatting this data with &#39;%d&#39; instead.